susemanager-utils/susemanager-sls/modules/roster/uyuni.py

```diff
@@ -62,6 +62,9 @@ class UyuniRoster:
     """
 
     def __init__(self, db_config, uyuni_roster_config):
+        self.config_hash = hashlib.sha256(
+            str(uyuni_roster_config).encode(errors="backslashreplace")
```

Comment:
What's the type of `uyuni_roster_config`?

Reply:
It's a dictionary.

Reply:
Ah, ok. The order of the keys is stable, right?

Reply:
Order of keys is only stable from Python 3.6 onwards: https://docs.python.org/3/library/stdtypes.html#dict

Reply:
But I guess we only have Python 3.6+ anyway, so we should be safe.

Reply:
Right, 3.6 is the only version we are using for Uyuni/4.2/4.1 on the master side where this module could be invoked.

Reply:
I was actually thinking about whatever code path adds the stuff to the dict (let's image something parses a YAML file and randomly sorts it before adding the contents to `uyuni_roster_config`), but I agree - the worst case is not that bad anyway.

Reply:
The source of this dict actually a `__opts__` from the master (the same for salt-api): `__opts__.get("uyuni_roster")`
It's defined in the master's config.